pkg/controllers/binding/common.go

```diff
@@ -157,18 +157,20 @@ func mergeLabel(workload *unstructured.Unstructured, binding metav1.Object, scop
 	util.MergeLabel(workload, util.ManagedByKarmadaLabel, util.ManagedByKarmadaLabelValue)
 	if scope == apiextensionsv1.NamespaceScoped {
 		bindingID := util.GetLabelValue(binding.GetLabels(), workv1alpha2.ResourceBindingPermanentIDLabel)
-		util.MergeLabel(workload, workv1alpha2.ResourceBindingReferenceKey, names.GenerateBindingReferenceKey(binding.GetNamespace(), binding.GetName()))
 		util.MergeLabel(workload, workv1alpha2.ResourceBindingPermanentIDLabel, bindingID)
+		workLabel[workv1alpha2.ResourceBindingPermanentIDLabel] = bindingID
 
+		// TODO(whitewindmills): Delete the following two lines in a future version.
 		workLabel[workv1alpha2.ResourceBindingReferenceKey] = names.GenerateBindingReferenceKey(binding.GetNamespace(), binding.GetName())
-		workLabel[workv1alpha2.ResourceBindingPermanentIDLabel] = bindingID
+		util.MergeLabel(workload, workv1alpha2.ResourceBindingReferenceKey, names.GenerateBindingReferenceKey(binding.GetNamespace(), binding.GetName()))
```

Comment:
a future version, do you mean v1.11+ (not in v1.10)?

Reply:
yes